Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2491,11 +2491,30 @@ bool RenderElement::createsNewFormattingContext() const
 bool RenderElement::establishesIndependentFormattingContext() const
 {
     auto& style = this->style();
-    return isFloatingOrOutOfFlowPositioned()
-        || (isBlockBox() && hasPotentiallyScrollableOverflow())
+    auto hasPaintContainment = [&] {
+        if (auto* element = this->element())
+            return style.usedContain().contains(Containment::Paint) && WebCore::shouldApplyPaintContainment(style, *element);
```

Comment:
`shouldApplyPaintContainment` does the previous check already. Though is there a reason not to use the bit flag?

Reply:
The idea is going to be to allow `establsiesIndependetFormattingContext` to take in any style and compute a result based off that. That somewhat begs that question if it should then just live on `RenderStyle` (probably not?) but I'll leave that for a more open discussion :)